### ISSUE TITLE
SQL file with multiple "for" loops throws exceptions in L016

### ISSUE THREAD

**Issue by barrywhart** [sqlfluff/sqlfluff#1433]
## Expected Behaviour

Successful run of `sqlfluff fix`.

## Observed Behaviour

Multiple exceptions like the following:
```
Traceback (most recent call last):
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/rules/base.py", line 311, in crawl
    templated_file=templated_file,
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/rules/L016.py", line 503, in _eval
    fixes = self._eval_line_for_breaks(this_line)
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/rules/L016.py", line 266, in _eval_line_for_breaks
    raise ValueError("Is this possible?")  # pragma: no cover
ValueError: Is this possible?
CRITICAL   [L016] Applying rule L016 threw an Exception: Is this possible?
```

## Steps to Reproduce

Run `sqlfluff fix` on this `test.sql` file:
```
-- This file combines product data from individual brands into a staging table
{% set products =  [
  'table1',
  'table2'] %}

{% for product in products %}
SELECT
  brand,
  country_code,
  category,
  name,
  id
FROM
  {{ product }};
{% endfor %}

{% for product in products %}
SELECT
  brand,
  country_code,
  category,
  name,
  id
FROM
  {{ product }};
{% endfor %}
```

## Dialect
`ansi`

## Version
SQLFluff version: `main` as of September 22 (commit hash `f3091db9c887a62eb7eb6b6437a8417122420ffa`)
Python version: 3.7.9

## Configuration
n/a

**Comment by tunetheweb** [comment]
@barrywhart could you retest this? Suspect it’s fixed with #1494

**Comment by barrywhart** [comment]
Confirmed -- works now.
